QUESTION: Graceful way to check for several things on startup

@kevinpapst I was trying to find a question template for this, as this is not a bug. I am currently working on an nginx-based docker image. However, I want to make the deployment as hassle free as possible, controlled by the docker ENV vars.

I am using a base image that allows me to specify a number of entry point scripts (to be executed on launch of the container). Everything works, but after the first time, as the script tries to create schema, migrations, admin user (specified in docker-compose) it is obviously going to fail. The container works just fine,  though.

However, the schema:create command clearly specifies "do not run this on a production environment", I would assume because it might break it?

Therefore, I am looking for a graceful way to check for:
* Database exists.
* Update schema exists.
* Migration is up to date.
* First superadmin (local) user exists.

This, using preferable a simple bash script, but I am open to a php script as well.

I think I have also found a way to send the output of the current log (env/prod) to the stdout. The base image I have uses supervisor. If I put the tail of of the log into a supervisor job, it will live for the duration of the container, and will display wherever the container's stdout is shown.

Trying to use the kimai:install method but I get this in my container's stdout:
```
web_1  | Welcome to the interactive Kimai installer!
web_1  | ===========================================
web_1  |
web_1  |  [ERROR] Installation only works in interactive mode
```

--- a/src/Command/InstallCommand.php
+++ b/src/Command/InstallCommand.php
@@ -88,7 +88,6 @@ class InstallCommand extends Command
         // $io->note(sprintf('You have chosen the "%s" environment', $environment));
         $environment = getenv('APP_ENV');
 
-        // create database if necessary
         try {
             $this->createDatabase($io, $input, $output);
         } catch (\Exception $ex) {
@@ -126,12 +125,8 @@ class InstallCommand extends Command
     protected function rebuildCaches(string $environment, SymfonyStyle $io, InputInterface $input, OutputInterface $output)
     {
         if ($input->isInteractive()) {
-            if (!$this->askConfirmation(
-                $input,
-                $output,
-                'Do you want me to rebuild the caches (yes) or skip this step (no)?',
-                true
-            )) {
+            $question = 'Do you want me to rebuild the caches (yes) or skip this step (no)?';
+            if (!$this->askConfirmation($input, $output, $question, true)) {
                 return;
             }
         }

--- a/src/Command/InstallCommand.php
+++ b/src/Command/InstallCommand.php
@@ -217,6 +217,8 @@ class InstallCommand extends Command
             return self::ERROR_PERMISSIONS;
         }
         $io->writeln('');
+
+        return true;
     }
 
     protected function importMigrations(SymfonyStyle $io, OutputInterface $output)

--- a/src/Command/InstallCommand.php
+++ b/src/Command/InstallCommand.php
@@ -76,16 +76,9 @@ class InstallCommand extends Command
     {
         $io = new SymfonyStyle($input, $output);
 
-        $io->title('Welcome to the interactive Kimai installer!');
+        $io->title('Kimai installer - v' . Constants::VERSION);
 
-        if (!$input->isInteractive()) {
-            $io->error('Installation only works in interactive mode');
-
-            return self::ERROR_INTERACTIVE;
-        }
-
-        $rows = $this->checkPermissions();
-        $result = $this->confirmAbortToReviewPermissions($io, $input, $output, $rows);
+        $result = $this->reviewPermissions($io, $input, $output);
         if (true !== $result) {
             return $result;
         }
@@ -132,8 +125,15 @@ class InstallCommand extends Command
 
     protected function rebuildCaches(string $environment, SymfonyStyle $io, InputInterface $input, OutputInterface $output)
     {
-        if (!$this->askConfirmation($input, $output, 'Do you want me to rebuild the caches (yes) or skip this step (no)?', true)) {
-            return;
+        if ($input->isInteractive()) {
+            if (!$this->askConfirmation(
+                $input,
+                $output,
+                'Do you want me to rebuild the caches (yes) or skip this step (no)?',
+                true
+            )) {
+                return;
+            }
         }
 
         $io->text('Rebuilding your cache now, please be patient ...');
@@ -196,8 +196,14 @@ class InstallCommand extends Command
         return $rows;
     }
 
-    protected function confirmAbortToReviewPermissions(SymfonyStyle $io, InputInterface $input, OutputInterface $output, array $permissions)
+    protected function reviewPermissions(SymfonyStyle $io, InputInterface $input, OutputInterface $output)
     {
+        if (!$input->isInteractive()) {
+            return true;
+        }
+
+        $permissions = $this->checkPermissions();
+
         if (empty($permissions)) {
             return true;
         }

